core/sql/optimizer/BindRI.cpp

```diff
@@ -417,12 +450,18 @@ void RefConstraint::getPredicateText(NAString &text,
 				     NAString *corrName) const
 {
   NAString tblText = ( (corrName == NULL) ? tblName.getQualifiedNameAsAnsiString() : *corrName);
+  int pos= 0;
   text += "(";
   for (CollIndex i = 0; i < keyColumns.entries(); i++)
     {
-      if (i)
+      if(isHiddenColumn(keyColumns[i]->getColName()) )
```

Comment:
Yes, I agree. There are more methods in NAColumn, These names can change and users can choose their own name for the division columns if they like (although I would not recommend that). Why are we excluding salt and division here? It would be best to choose the right one of the many methods in NAColumn:

  isUserColumn()
  isSystemColumn()
  isSyskeyColumn()
  isIdentityColumn()
  isIdentityColumnByDefault()
  isIdentityColumnAlways()
  isComputedColumn()
  isComputedColumnAlways()